### DISCUSSION TITLE
Emma Integration: activePower

### DISCUSSION THREAD

**Discussion by bolliy** [bolliy/ioBroker.sun2000#124] [Allgemein]
Über das issue https://github.com/bolliy/ioBroker.sun2000/issues/63 wurde ein Anregung von @cksoft0807 geteilt, die ich gerne hier zu Diskussion stellen möchte:
"Ich würde gerne noch einmal beim Thema "meter/activePower" einhaken. Wäre nicht eigentlich der Wert Sampled Data, "Feed-in power" (30358) das passende Register für die Anlage? Müsste dann auch das Problem interner/externer Sensor lösen.

Die Hauslast sollte man mit Emma eigentlich auch ohne Berechnung über "Load power, 30356" abgreifen können?"

LG Stephan

**Comment by bolliy** [comment]
Der Wert  "Feed-in power" würde u.U. als "meter/activePower" funktionieren. Nur sind im path meter noch weitere Leistungsdaten, die ggf. nicht als kummulierten Wert (Sampled data) bekomme. So müsste ich für den externen Sensor doch einbinden. 

Wer den Wert emma.loadPower lieber als sun2000.0.collected.houseConsumption verwenden möchte, kann dieses natürlich tun.
Interessant wäre ob die beiden Werte nahezu gleich sind? Die Frage stellt sich natürlich auch für die Werte meter/activePower und emma.feed-inPower?

Ich bin gespannt wie ihr das sieht?
LG Stephan

**Comment by cksoft0807** [comment]
Ich habe genau das Problem von dir @bolliy , keine Emma und die Implementierung dieser in eine App.

Meiner Meinung nach sollten die Register
30330 – Feed-in to grid today = Einspeisung heute?
30332 – Total feed-in to grid = Einspeisung total?
30336 - Supply from grid today = Bezug heute?
30338 – Total supply from grid = Bezug total?
30358 – Feed-in power = aktuelle Leistung Bezug/Einspeisung?
denen unter meter/ intern/extern entsprechen. Je nachdem was verbunden ist. Zudem wäre interessant ob die Vorzeichen stimmen.

Außerdem gibt es mit der Emma neu diese Register. Sofern die Daten konsistent wären könnte man sich entsprechende Berechnungen sparen.
30324 – Consumption today = Hausverbrauch heute
30326 – Total energy consumption = Hausverbrauch total
30356 – Load power = aktuell benötigte Leistung im Haus
30346 – PV yield today = PV-Modul Ertrag heute
30348 – Total PV energy yield = PV-Modul Ertrag total

Was sich mir ebenfalls nicht erschließt sind die Register
30318 - ESS chargeable energy = noch freie kWh aller Batterien unter Berücksichtigung der eingestellten SOC Grenzen?
30320 – ESS dischargeable enrgy = Batterieinhalt unter Berücksichtigung der SOC Grenzen?
30322 – Rated ESS capacity = Batteriespeicher ohne Berücksichtigung der SOC Grenzen?
30369 – ESS chargeable capacity = noch freie kWh aller Batterien ohne Berücksichtigung der SOC Grenzen?
30371 – ESS dischargeable capacity = Batterieinhalt ohne Berücksichtigung der SOC Grenzen?

Wäre wirklich klasse, wenn Emma Besitzer uns hier helfen könnten.

**Reply by bolliy** [reply]
@cksoft0807 an welcher App entwickelst du gerade? Kommerziell oder non-kommerziell?
...

**Reply by bolliy** [reply]
vielleicht ist das für dich interessant: https://github.com/Roving-Ronin/huawei-solar-lib/blob/dfd6f5dce7d20052c3a35e01994bfa2c1a866950/src/huawei_solar/registers.py#L1302

**Reply by cksoft0807** [reply]
Das ist als PVvis kostenlos in den Stores zu finden. Daran wird sich auch nichts ändern. ;-)

Danke für den Link. Die Register sind grundsätzlich schon bekannt, aber die Beschreibung oder auch der Inhalt in unterschiedlichen Konstellationen leider nicht immer eindeutig.

**Reply by bolliy** [reply]
Dein App schau ich mir gerne mal an ...
